src/server/federatedConnections/exchange.ts

```diff
@@ -0,0 +1,183 @@
+import * as oauth from "oauth4webapi"
+
+import {
+  FederatedConnectionAccessTokenErrorCode,
+  FederatedConnectionsAccessTokenError,
+  SdkError,
+} from "../../errors"
+import { TokenSet } from "../../types"
+import {
+  AuthServerMetadata,
+  MetadataDiscoverOptions,
+} from "../authServerMetadata"
+import { FederatedConnectionTokenSet } from "./serializer"
+
+/**
+ * A constant representing the grant type for federated connection access token exchange.
+ *
+ * This grant type is used in OAuth token exchange scenarios where a federated connection
+ * access token is required. It is specific to Auth0's implementation and follows the
+ * "urn:auth0:params:oauth:grant-type:token-exchange:federated-connection-access-token" format.
+ */
+export const GRANT_TYPE_FEDERATED_CONNECTION_ACCESS_TOKEN: string =
+  "urn:auth0:params:oauth:grant-type:token-exchange:federated-connection-access-token"
+
+/**
+ * Constant representing the subject type for a refresh token.
+ * This is used in OAuth 2.0 token exchange to specify that the token being exchanged is a refresh token.
+ *
+ * @see {@link https://tools.ietf.org/html/rfc8693#section-3.1 RFC 8693 Section 3.1}
+ */
+export const SUBJECT_TYPE_REFRESH_TOKEN: string =
+  "urn:ietf:params:oauth:token-type:refresh_token"
+
+/**
+ * A constant representing the token type for federated connection access tokens.
+ * This is used to specify the type of token being requested from Auth0.
+ *
+ * @constant
+ * @type {string}
+ */
+export const REQUESTED_TOKEN_TYPE_FEDERATED_CONNECTION_ACCESS_TOKEN: string =
+  "http://auth0.com/oauth/token-type/federated-connection-access-token"
+
+export type FederatedConnectionTokenExchangeOptions = {
+  /**
+   * The set of tokens to be exchanged.
+   */
+  tokenSet: TokenSet
+  /**
+   * The name of the federated connection.
+   */
+  connection: string
+  /**
+   * Optional hint for login.
+   */
+  login_hint?: string,
+  existingTokenSet?: FederatedConnectionTokenSet
```

Comment:
Can we rename this to `existingFederatedConnectionTokenSet` and add some jsdocs?